### PR TITLE
chore: add API for edit user profile (#4668)

### PR DESCRIPTION
**Pull request by tofumatt**
Gonna do the API/reducer/saga stuff for this all separately and maybe it'll make the PRs nice and easy.

Here's the API and tests for editing a user profile. I've exclude the image upload for now as I'll do that last.

Part of #4668.

### CODE REVIEW

File: tests/unit/amo/api/test_users.js [hunk 1]
```diff
@@ -30,16 +38,46 @@ describe(__filename, () => {
...
+  describe('editUserAccount', () => {
+    const mockResponse = (newParams) => createApiResponse({
+      jsonData: createUserAccountResponse(newParams),
     });
 
-    it('throws an error if api state is missing', () => {
-      expect(() => {
-        currentUserAccount({});
-      }).toThrowError(/api state is required/);
+    const getParams = (params = {}) => {
```
Comment (bobsilverberg) [inline tests/unit/amo/api/test_users.js]: This is just a nit; take it or leave it. Considering that there's only one test that makes use of it, this `getParams` function is kind of unnecessary. It does seem like a good model to follow, and I know we use it in other test files.  I notice the same model is followed for the other tests in this file, which also don't really need it. If  more tests are added later it would be useful, but at this point none of these cases seems to need more tests.
